sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+    private readonly casService: CASService,
+    @InjectRepository(CASInvoice)
+    private readonly casInvoiceRepo: Repository<CASInvoice>,
+    private readonly dataSource: DataSource,
+  ) {}
+  /**
+   * Get the list of pending invoices to be sent from the approved batch.
+   * @returns list of pending invoices.
+   */
+  async getPendingInvoices(): Promise<CASInvoice[]> {
+    const pendingInvoices = await this.dataSource
+      .getRepository(CASInvoice)
+      .find({
+        select: {
+          id: true,
+          casSupplier: {
+            id: true,
+            supplierNumber: true,
+            supplierAddress: { supplierSiteCode: true },
+          },
+          createdAt: true,
+          invoiceNumber: true,
+          disbursementReceipt: {
+            id: true,
+            fileDate: true,
+            createdAt: true,
+          },
+          casInvoiceBatch: {
+            id: true,
+            batchName: true,
+            batchDate: true,
+            createdAt: true,
+          },
+          casInvoiceDetails: {
+            id: true,
+            valueAmount: true,
+            casDistributionAccount: {
+              id: true,
+              operationCode: true,
+              awardValueCode: true,
+              distributionAccount: true,
+            },
+          },
+        },
+        relations: {
+          casInvoiceBatch: true,
+          disbursementReceipt: true,
+          casSupplier: true,
+          casInvoiceDetails: {
+            casDistributionAccount: true,
+          },
+        },
+        where: {
+          casInvoiceBatch: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+          invoiceStatus: CASInvoiceStatus.Pending,
+        },
+      });
+    return pendingInvoices;
+  }
+
+  /**
+   * Send pending invoices to CAS.
+   * @param parentProcessSummary parent process summary for logging.
+   * @param pendingInvoices pending invoices.
+   */
+
+  async sendInvoices(
+    parentProcessSummary: ProcessSummary,
+    pendingInvoices: CASInvoice[],
+  ): Promise<void> {
+    // Process each invoice in parallel.
+    await processInParallel(
+      (pendingInvoice) =>
+        this.processInvoice(pendingInvoice, parentProcessSummary),
+      pendingInvoices,
+    );
+  }
+
+  /**
+   * Process and send an invoice to CAS.
+   * @param pendingInvoice pending invoice.
+   * @param parentProcessSummary parent process summary for logging.
+   */
+  private async processInvoice(
+    pendingInvoice: CASInvoice,
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    // Log information about the pending invoice being processed.
+    parentProcessSummary.info(
+      `Processing pending invoice: ${pendingInvoice.invoiceNumber}.`,
+    );
+    const pendingInvoicePayload = this.getPendingInvoicePayload(pendingInvoice);
+    parentProcessSummary.info(
+      `Pending invoice payload invoice number: ${pendingInvoicePayload.invoiceNumber}.`,
+    );
```

Comment:
Is this log message required? I am not able to follow the meaning. Can you please clarify or remove it?